ktlint-core/src/main/kotlin/com/pinterest/ktlint/core/KtLint.kt

```diff
@@ -51,10 +54,18 @@ public object KtLint {
      * @param script true if this is a Kotlin script file
      * @param editorConfigPath optional path of the .editorconfig file (otherwise will use working directory)
      * @param debug True if invoked with the --debug flag
+     * @param editorConfigOverride should contain entries to add/replace from loaded `.editorconfig` files.
+     *
+     * For possible keys check related [Rule]s that implements [UsesEditorConfigProperties] interface.
+     *
+     * For values use `PropertyType.PropertyValue.valid("override", <expected type>)` approach.
+     * It is also possible to set value into "unset" state by using [PropertyType.PropertyValue.UNSET].
+     *
      * @param isInvokedFromCli **For internal use only**: indicates that linting was invoked from KtLint CLI tool.
      * Enables some internals workarounds for Kotlin Compiler initialization.
      * Usually you don't need to use it and most probably it will be removed in one of next versions.
      */
+    @OptIn(FeatureInAlphaState::class)
```

Comment:
I am thinking about introducing `ExperimentalParams` that will have `@FeatureInAlphaState` annotation.

`Params` will not have any experimental properties, but actually will create `ExperimentalParams` with default values for experimental properties.